test/e2e/tests/request-queuing/dapp1-send-dapp2-signTypedData.spec.js

```diff
@@ -110,10 +107,7 @@ describe('Request Queuing Dapp 1, Switch Tx -> Dapp 2 Send Tx', function () {
           `window.ethereum.request(${switchEthereumChainRequest})`,
         );
 
-        await driver.waitUntilXWindowHandles(4);
-        await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
-
-        await driver.clickElement({ text: 'Switch network', tag: 'button' });
```

Comment:
After that is fixed, we can see a new error: in the test dapp 0, we are switching networks from Mainnet to `0x3e8`,  but when we perform the send from the test dapp 0, the transaction shows Mainnet. I think that's an issue on the wallet level


https://github.com/user-attachments/assets/35fb43d1-5acc-4fee-b73c-9c36b11d4961

Reply:
I tried to replicate it in chrome and it's working there

Reply:
@seaona the situation you're describing sounds similar to what I saw here: https://github.com/MetaMask/metamask-extension/pull/27561#discussion_r1787672800

In both cases, the confirmation has the wrong chain

Reply:
Thanks @Gudahtt @seaona , I have pushed this commit that hopefully fix the race condition. https://github.com/MetaMask/metamask-extension/pull/27561/commits/5dbdc39dcb5951b5d53fe8de771cd4b97854695f